e2e/websocket_test.go

```diff
@@ -0,0 +1,161 @@
+package e2e
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/0xPolygon/minimal/e2e/framework"
+	"github.com/0xPolygon/minimal/jsonrpc"
+	"github.com/0xPolygon/minimal/types"
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
+	"github.com/umbracle/go-web3"
+)
+
+type testWSRequest struct {
+	JSONRPC string   `json:"jsonrpc"`
+	Params  []string `json:"params"`
+	Method  string   `json:"method"`
+	ID      int      `json:"id"`
+}
+
+func constructWSRequest(id int, method string, params []string) ([]byte, error) {
+	request := testWSRequest{
+		JSONRPC: "2.0",
+		Method:  method,
+		ID:      id,
+		Params:  params,
+	}
+
+	return json.Marshal(request)
+}
+
+func getWSResponse(t *testing.T, ws *websocket.Conn, request []byte) (jsonrpc.Response, error) {
+	if wsError := ws.WriteMessage(websocket.TextMessage, request); wsError != nil {
+		t.Fatalf("Unable to write message to WS connection: %v", wsError)
+	}
+	_, response, wsError := ws.ReadMessage()
+	if wsError != nil {
+		t.Fatalf("Unable to read message from WS connection: %v", wsError)
+	}
+
+	var res jsonrpc.Response
+	if wsError = json.Unmarshal(response, &res); wsError != nil {
+		t.Fatalf("Unable to unmarshal WS response: %v", wsError)
+	}
+
+	return res, wsError
+}
+
+func TestWS_Response(t *testing.T) {
+	preminedAccounts := []struct {
+		address types.Address
+		balance *big.Int
+	}{
+		{types.StringToAddress("1"), big.NewInt(10)},
+		{types.StringToAddress("2"), big.NewInt(20)},
+	}
+
+	srvs := framework.NewTestServers(t, 1, func(config *framework.TestServerConfig) {
+		config.SetConsensus(framework.ConsensusDev)
+		config.SetSeal(true)
+
+		for _, account := range preminedAccounts {
+			config.Premine(account.address, account.balance)
+		}
+	})
+	srv := srvs[0]
+	client := srv.JSONRPC()
+
+	// Convert the default JSONRPC address to a WebSocket one
+	wsURL := "ws" + strings.TrimPrefix(srv.JsonRPCAddr(), "http") + "/ws"
+
+	// Connect to the websocket server
+	ws, _, err := websocket.DefaultDialer.Dial(wsURL, nil)
+	if err != nil {
+		t.Fatalf("Unable to connect to WS: %v", err)
+	}
+	defer ws.Close()
+
+	t.Run("Valid account balance", func(t *testing.T) {
+		requestID := 1
+		request, constructErr := constructWSRequest(
+			requestID,
+			"eth_getBalance",
+			[]string{preminedAccounts[0].address.String(), "latest"},
+		)
+
+		assert.Nilf(t, constructErr, fmt.Sprintf("Unable to construct request: %v", constructErr))
```

Comment:
Might be better to do `t.Fatalf` here, since you might want the test to stop if this happens
```suggestion
		if constructErr != nil {
			t.Fatalf("Unable to construct request: %v", constructErr)
		}
```

Reply:
Fixed 👍